ambry-router/src/main/java/com.github.ambry.router/DeleteManager.java

```diff
@@ -102,6 +106,7 @@ void submitDeleteBlobOperation(String blobIdString, FutureResult<Void> futureRes
       DeleteOperation deleteOperation =
           new DeleteOperation(routerConfig, routerMetrics, responseHandler, blobId, futureResult, callback, time);
       deleteOperations.add(deleteOperation);
+      readyForPollCallback.onPollReady();
```

Comment:
Yes. But will have this done right after operation submission in the caller code as you suggested in one of the other comments.